instance/models/openedx_instance.py

```diff
@@ -124,17 +125,40 @@ def domain_slug(self):
         prefix = ('edxins-' + slugify(self.domain))[:20]
         return "{prefix}-{num}".format(prefix=prefix, num=self.id)
 
+    def get_domain(self, site):
+        """
+        Returns the external domain for the given site if present; otherwise, falls back to the internal domain.
+        """
+        internal_attr = 'internal_{}_domain'.format(site)
+        external_attr = 'external_{}_domain'.format(site)
+        return getattr(self, external_attr, None) or getattr(self, internal_attr)
```

Comment:
It's a bit of a hack that the field names now define the structure of the data.  In general, it's not a good idea to put data in variable names.

An alternative would be to have a field `domains`, which stores a JSON dictionary mapping site names to lists `["external.domain", "internal.domain"]`, or a similar construct.  However, since the version of Postgres we are using does not support JSONB fields yet, it's probably best to just leave it as you implemented it for now.